### ISSUE TITLE
Issue with using the document base for the default vocabulary in JSON-LD 1.1

### ISSUE THREAD

**Issue by Tomvbe** [apache/jena#2196]
### Version

4.10.0

### What happened?

**Description**
As of JSON-LD 1.1 it is possible to use "#" as the vocabulary mapping (https://www.w3.org/TR/json-ld11/#document-relative-vocabulary-mapping).

Jena parses this incorrectly, it ignores the "#".  

**Reproduction**
When we expand the following JSON-LD 1.1 to TURTLE:

```
{
  "@context": {
    "@version": 1.1,
    "@base": "http://example/document",
    "@vocab": "#"
  },
  "@id": "http://example.org/places#BrewEats",
  "@type": "Restaurant",
  "name": "Brew Eats"
}
```
in jena we get the following output (notice the missing '#' before 'Restaurant' and 'name'):

```
@prefix : <#> .

<http://example.org/places#BrewEats> a <http://example/documentRestaurant>;
    <http://example/documentname>  "Brew Eats" .
```

We expect the following output (cfr. https://www.w3.org/TR/json-ld11/#document-relative-vocabulary-mapping):
```
<http://example.org/places#BrewEats> a <http://example/document#Restaurant> ;
    <http://example/document#name> "Brew Eats" .
```

### Are you interested in making a pull request?

None

**Comment by filip26** [comment]
I confirm it's expansion bug. An input is expanded before serialization into RDF so the bug propagates down the stream. Will be fixed.

**Comment by filip26** [comment]
Fixed in trunk (`1.4.0-SNAPSHOT`).  Titanium `v1.4.0` will be released during February, hopefully.

**Comment by filip26** [comment]
just heads up, v1.4.0 has been released https://github.com/filip26/titanium-json-ld/releases/tag/v1.4.0
